components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -179,7 +179,7 @@ impl DedicatedWorkerGlobalScope {
                 destination: Destination::Worker,
                 credentials_mode: CredentialsMode::Include,
                 use_url_credentials: true,
-                origin: worker_url,
+                origin: GlobalScope::current().origin().immutable().clone(),
```

Comment:
This is probably broken - we don't have a current global in the new thread yet, so we need to clone the origin before spawning the thread and use that value instead.